pkg/neonvm/controllers/vm_controller.go

```diff
@@ -802,6 +837,217 @@ func (r *VMReconciler) doReconcile(ctx context.Context, vm *vmv1.VirtualMachine)
 	return nil
 }
 
+func (r *VMReconciler) reconcileCertificateSecretIfNeeded(ctx context.Context, vm *vmv1.VirtualMachine) (*corev1.Secret, error) {
```

Comment:
random idea: WDYT about putting all of these certificate related functions in a separate file? There's a lot here, and I figure it might be a good opportunity to not let this humongous file grow any larger :sweat_smile: